src/editions/SingleBatchEditionFactory.sol

```diff
@@ -23,49 +45,54 @@ contract SingleBatchEditionCreator {
     string internal constant SYMBOL = unicode"✦ SHOWTIME";
 
     address public immutable editionImpl;
-    address public immutable minter;
     IShowtimeVerifier public immutable showtimeVerifier;
 
     /// @param _editionImpl the address of the implementation contract for SingleBatchEdition
-    constructor(address _editionImpl, address _minter) {
-        if (_editionImpl == address(0) || _minter == address(0)) {
+    constructor(address _editionImpl, address _showtimeVerifier) {
+        if (_editionImpl == address(0)) {
             revert NullAddress();
         }
 
         editionImpl = _editionImpl;
-        minter = _minter;
-        showtimeVerifier = IGatedEditionMinter(_minter).showtimeVerifier();
+        showtimeVerifier = IShowtimeVerifier(_showtimeVerifier);
     }
 
-    /// Creates a new edition contract as a factory with a deterministic address
-    /// @param name Name of the edition contract
-    /// @param description Metadata: Description of the edition entry
-    /// @param animationUrl Metadata: Animation url (optional) of the edition entry
-    /// @param imageUrl Metadata: Image url (semi-required) of the edition entry
-    /// @param royaltyBPS BPS amount of royalties
-    /// @param signedAttestation the attestation to verify along with a corresponding signature
+    /// Creates and mint a new SingleBatchEdition contract with a deterministic address
     /// @dev we expect the signed attestation's context to correspond to this contract's address
     /// @dev we expect the signed attestation's beneficiary to be the edition's creator
+    /// @param pointer the address of the SSTORE2 pointer with the recipients of the batch mint for this edition
+    /// @param signedAttestation a signed message from Showtime authorizing this action on behalf of the edition creator
     /// @return edition the address of the created edition
     function createEdition(
-        string calldata name,
-        string calldata description,
-        string calldata animationUrl,
-        string calldata imageUrl,
-        uint256 royaltyBPS,
-        string calldata externalUrl,
-        string calldata creatorName,
-        string calldata tags,
+        EditionData calldata data,
+        address pointer,
         SignedAttestation calldata signedAttestation
-    ) external returns (ISingleBatchEdition edition) {
+    ) public returns (ISingleBatchEdition edition) {
         validateAttestation(signedAttestation);
         address creator = signedAttestation.attestation.beneficiary;
 
-        bytes32 salt = keccak256(abi.encodePacked(creator, name, animationUrl, imageUrl));
+        bytes32 salt = keccak256(abi.encodePacked(creator, data.name, data.animationUrl, data.imageUrl));
+        address predicted = address(getEditionAtId(uint256(salt)));
```

Comment:
So `getEditionAtId` returns 0 if it doesn't exist?

Reply:
no, it returns the computed address, whether it exists or not

Reply:
Ohh i missed the `.code` part [here](https://github.com/showtime-xyz/showtime-contracts/blob/atomic-batch-mints/src/editions/SingleBatchEditionFactory.sol#L78)

So that's checking `if the code deployed at this address has a length of 0` right?

Reply:
yes!